Tools/AP_Periph/AP_Periph.h

```diff
@@ -142,6 +160,19 @@ class AP_Periph_FW {
     void rcout_handle_safety_state(uint8_t safety_state);
 #endif
 
+#if defined(HAL_PERIPH_ENABLE_NOTIFY) || defined(HAL_PERIPH_NEOPIXEL_COUNT)
+    // notification object for LEDs, buzzers etc (parameter set to
+    // false disables external leds)
+    struct AP_Periph_Notify {
+#ifdef HAL_PERIPH_ENABLE_NOTIFY
+        AP_Notify lib;
+#endif
+        uint32_t update_ms;
+
+        void update_rainbow();
```

Comment:
what is the reason for defining this under the structure?

Reply:
It's LED related. In general, we're trying to group all the features into classes for better organization.

Reply:
removed the struct